entrypoint

```diff
@@ -1,9 +1,16 @@
 #! /usr/bin/env bash
 
-# When deploying the dashboard charm we need to update the configurations whenever 
-# the relation with the controller changes, so they are generated by the charm
-# and mouted into `/srv/configs` on the container
-cp /srv/configs/config.js /srv/config.js 2>/dev/null
-cp /srv/configs/nginx.conf /etc/nginx/sites-available/default 2>/dev/null
+# When deploying the dashboard charm the dashboard and nginx configs are generated
+# by the charm. These configs can also be set when launching the container using environment variables.
+# The following environment variables can be provided. Only DASHBOARD_CONTROLLER_URL is required.
+# DASHBOARD_CONTROLLER_URL - equivalent to the charm's controller-url config option.
+# DASHBOARD_ROOT - the root directory that the dashboard is located in.
+# DASHBOARD_CONFIG_DIR - the location that the config.js.j2 and nginx.conf.j2 are located in.
+# DASHBOARD_IDENTITY_PROVIDER_URL - equivalent to the charm's identity-provider-url config option.
+# DASHBOARD_IS_JUJU - equivalent to the charm's is-juju config option.
+# DASHBOARD_ANALYTICS_ENABLED - equivalent to the charm's analytics-enabled config option.
+# DASHBOARD_PORT - equivalent to the charm's port config option.
+
+python3 /srv/config.py
 
 nginx -g "daemon off;"
```

Comment:
I noticed that when I would do `docker compose down`, the dashboard container wouldn't exit immediately, only after 10 seconds. From docker docs,
>When executing docker compose down, Docker Compose by default sends a SIGTERM signal to containers, allowing them a grace period to shut down gracefully. If a container does not terminate within this grace period, a SIGKILL signal is sent to forcefully stop it. The default grace period for docker compose down is 10 seconds.

There is an easy fix. The common thing is to do `exec <your app>` in the entrypoint see https://stackoverflow.com/questions/32255814/what-purpose-does-using-exec-in-docker-entrypoint-scripts-serve.

Reply:
Fixed.